Sources/Optimizely+Decide/OptimizelyUserContext.swift

```diff
@@ -119,18 +132,126 @@ public class OptimizelyUserContext {
     public func trackEvent(eventKey: String,
                            eventTags: OptimizelyEventTags? = nil) throws {
         
-        guard let optimizely = self.optimizely, optimizely.config != nil else {
+        guard let optimizely = self.optimizely else {
             throw OptimizelyError.sdkNotReady
         }
-
+        
         try optimizely.track(eventKey: eventKey,
                              userId: userId,
                              attributes: attributes,
                              eventTags: eventTags)
     }
+    
+}
 
+// MARK: - ForcedDecisions
+
+extension OptimizelyUserContext {
+    
+    /// Sets the forced decision (variation key) for a given flag and an optional rule.
+    /// - Parameters:
+    ///   - flagKey: A flag key.
+    ///   - ruleKey: An experiment or delivery rule key (optional).
+    ///   - variationKey: A variation key.
+    /// - Returns: true if the forced decision has been set successfully.
+    public func setForcedDecision(flagKey: String,
+                                  ruleKey: String? = nil,
+                                  variationKey: String) -> Bool {
+        
+        guard optimizely?.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return false
+        }
+        
+        if let index = forcedDecisions.firstIndex(where: { $0.flagKey == flagKey && $0.ruleKey == ruleKey }) {
+            forcedDecisions[index].variationKey = variationKey
+        } else {
+            forcedDecisions.append(ForcedDecision(flagKey: flagKey, ruleKey: ruleKey, variationKey: variationKey))
+        }
+        
+        return true
+    }
+    
+    /// Returns the forced decision for a given flag and an optional rule.
+    /// - Parameters:
+    ///   - flagKey: A flag key.
+    ///   - ruleKey: An experiment or delivery rule key (optional).
+    /// - Returns: A variation key or nil if forced decisions are not set for the parameters.
+    public func getForcedDecision(flagKey: String, ruleKey: String? = nil) -> String? {
+        guard optimizely?.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return nil
+        }
+        
+        return findForcedDecision(flagKey: flagKey, ruleKey: ruleKey)
+    }
+    
+    /// Removes the forced decision for a given flag and an optional rule.
+    /// - Parameters:
+    ///   - flagKey: A flag key.
+    ///   - ruleKey: An experiment or delivery rule key (optional).
+    /// - Returns: true if the forced decision has been removed successfully.
+    public func removeForcedDecision(flagKey: String, ruleKey: String? = nil) -> Bool {
+        guard optimizely?.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return false
+        }
+        
+        if let index = forcedDecisions.firstIndex(where: { $0.flagKey == flagKey && $0.ruleKey == ruleKey }) {
```

Comment:
Thoughts on implementing using a map of maps for storing the forcedDecisions? Lookup time would be quicker and we can avoid looping through elements.

'Map<String, Map<String, ForcedDecision>> forcedDecisions = new.......'

then lookup just using the two keys: 'forcedDecisions.get(flagKey).get(ruleKey)'